logdb/logdb_bench_test.go

```diff
@@ -0,0 +1,345 @@
+		t.Run(tt.name, func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				want, err := tt.prepare()
+				if err != nil {
+					b.Fatal(err)
+				}
+				got, err := db.NewestBlockID()
+				if err != nil {
+					b.Fatal(err)
+				}
+				assert.Equal(b, want, got)
+			}
+		})
+	}
+}
+
+// BenchmarkFakeDB_WriteBlocks creates a temporary database, performs some write only benchmarks and then deletes it
+func BenchmarkFakeDB_WriteBlocks(t *testing.B) {
+	dbPath, err := createTempDBPath()
+	defer os.Remove(dbPath)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	db, err := logdb.New(dbPath)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+
+	b := new(block.Builder).Build()
+
+	t.ResetTimer()
+	for i := 0; i < t.N; i++ {
+		for i := 0; i < 100000; i++ {
+			b = new(block.Builder).
+				ParentID(b.Header().ID()).
+				Transaction(newTx()).
+				Transaction(newTx()).
+				Build()
+			receipts := tx.Receipts{newReceipt(), newReceipt()}
+
+			w := db.NewWriter()
+			if err := w.Write(b, receipts); err != nil {
+				t.Fatal(err)
+			}
+
+			if err := w.Commit(); err != nil {
+				t.Fatal(err)
+			}
+		}
+	}
+}
+
+// BenchmarkTestDB_HasBlockID opens a log.db file and measures the performance of the HasBlockID functionality of LogDB.
+// Running: go test -bench=BenchmarkTestDB_HasBlockID  -benchmem  github.com/vechain/thor/v2/logdb -dbPath /path/to/log.db
+func BenchmarkTestDB_HasBlockID(b *testing.B) {
+	if dbPath == "" {
+		b.Fatal("Please provide a dbPath")
+	}
+	db, err := logdb.New(dbPath)
+	if err != nil {
+		b.Fatal(err)
+	}
+	defer db.Close()
+
+	b0 := new(block.Builder).Build()
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		has, err := db.HasBlockID(b0.Header().ID())
+		if err != nil {
+			b.Fatal(err)
+		}
+		assert.False(b, has)
+	}
+}
+
+// BenchmarkTestDB_NewestBlockID opens a log.db file and measures the performance of the NewestBlockID functionality of LogDB.
+// Running: go test -bench=BenchmarkTestDB_NewestBlockID  -benchmem  github.com/vechain/thor/v2/logdb -dbPath /path/to/log.db
+func BenchmarkTestDB_NewestBlockID(b *testing.B) {
+	if dbPath == "" {
+		b.Fatal("Please provide a dbPath")
+	}
+	db, err := logdb.New(dbPath)
+	if err != nil {
+		b.Fatal(err)
+	}
+	defer db.Close()
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		_, err := db.NewestBlockID()
+		if err != nil {
+			b.Fatal(err)
+		}
+	}
+}
+
+// BenchmarkTestDB_FilterEvents opens a log.db file and measures the performance of the Event filtering functionality of LogDB.
+// Running: go test -bench=BenchmarkTestDB_FilterEvents  -benchmem  github.com/vechain/thor/v2/logdb -dbPath /path/to/log.db
+func BenchmarkTestDB_FilterEvents(b *testing.B) {
+	if dbPath == "" {
+		b.Fatal("Please provide a dbPath")
+	}
+	db, err := logdb.New(dbPath)
+	if err != nil {
+		b.Fatal(err)
+	}
+	defer db.Close()
+
+	eventCriteria1 := make([]*logdb.EventCriteria, 0, 1)
+	eventCriteria2 := make([]*logdb.EventCriteria, 0, 1)
+
+	addressCriteria := &logdb.EventCriteria{
```

Comment:
The criteria seems to be a empty responded one,  can you pick a real world use case? For example, the VTHO transfer event.